File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>

int main(int argc , string argv[])
{
    //if the number of command line argument is more or less than 2
    if(argc!=2)
    {
        printf("error\n");
        return(1);
    }
    //if exactly 2 command line argument is given
    else
    {
        //converting key argument from string to int
        int k = atoi(argv[1]);
        //prompt for user
        printf("plaintext:  ");
        string s = get_string();
        printf("ciphertext: ");
        //iterating to each character of string
        for(int i=0, n = strlen(s); i<n; i++)
        {
            //if the given character is letter
            if(isalpha(s[i]))
            {
                //if the character is in uppercase
                if(isupper(s[i]))
                {
                    //calculating encrypted character and printing it
                    int x = (((s[i]-65)+k)%26);
                    char c = x + 65;
                    printf("%c",c);
                }
                //if character is in lower case
                if(islower(s[i]))
                {
                    //calculating encrypted character and printing it
                    int x = (((s[i]-97)+k)%26);
                    char c = x+97;
                    printf("%c",c);
                }
            }
            //if the given character is not letter that means symbol or number
            else
            {
                printf("%c",s[i]);
            }
        }
        //printing a new line at the end and returning 0 to main     
        printf("\n");
        return(0);
    }
}

File: caesar.py
import cs50
import sys

def main():
    argument = len(sys.argv)
    if argument != 2:
        print("error")
        exit(1)
        
    else:
        key = int(sys.argv[1])
        print("plaintext: ",end="")
        text = cs50.get_string()
        print("ciphertext: ",end="")
        for c in text:
            if c.isalpha():
                if c.isupper():
                    x = ((ord(c)-65)+key)%26
                    letter = chr(x+65)
                    print(letter,end="")
                    
                if c.islower():
                    x = ((ord(c)-97)+key)%26
                    letter = chr(x+97)
                    print(letter,end="")
                    
            else:
                print(c,end="")
                
        print()
        
if __name__ == "__main__":
    main()
                
                    
                
    